### PR TITLE
small perf changes

### PR DESCRIPTION
**Pull request by orklah**
A list of small issues I noticed while working on Psalm PR. Mainly small perf optimisations. Those are reported by PHP EA on PHPStorm.

### CODE REVIEW

File: src/Roave/SecurityAdvisories/Version.php [hunk 1]
```diff
@@ -40,17 +41,17 @@ final class Version
      */
     private function __construct(array $matches)
     {
-        $this->versionNumbers = self::removeTrailingZeroes(...array_map('intval', explode('.', $matches['version'])));
+        $this->versionNumbers = self::removeTrailingZeroes(...array_map('\intval', explode('.', $matches['version'])));
```
Comment (Ocramius) [inline src/Roave/SecurityAdvisories/Version.php]: This is wrong: strings are already references to the FQN. The tool has a false positive here

File: src/Roave/SecurityAdvisories/Version.php [hunk 2]
```diff
@@ -40,17 +41,17 @@ final class Version
...
 
         $this->stabilityNumbers = [];
-        if (! isset($matches['stability_numbers'])) {
+        if (! array_key_exists('stability_numbers', $matches)) {
             return;
         }
 
         $this->stabilityNumbers = self::removeTrailingZeroes(
-            ...array_map('intval', explode('.', $matches['stability_numbers']))
+            ...array_map('\intval', explode('.', $matches['stability_numbers']))
```
Comment (Ocramius) [inline src/Roave/SecurityAdvisories/Version.php]: Same as above: wrong optimisation

File: src/Roave/SecurityAdvisories/VersionConstraint.php [hunk 3]
```diff
@@ -164,8 +164,8 @@ private function containsLowerBound(?Boundary $otherLowerBoundary) : bool
             return false;
         }
 
-        if (($this->lowerBoundary->limitIncluded() === $otherLowerBoundary->limitIncluded())
-            || $this->lowerBoundary->limitIncluded()
+        if ($this->lowerBoundary->limitIncluded()
```
Comment (Ocramius) [inline src/Roave/SecurityAdvisories/VersionConstraint.php]: I think the first expression should be put into a variable

File: src/Roave/SecurityAdvisories/Version.php (hunk 1 above)
Comment (orklah) [reply src/Roave/SecurityAdvisories/Version.php]: Wow, this is confusing! Good to know though. I'll revert that and create an issue for that FP

Comment (orklah) [reply src/Roave/SecurityAdvisories/Version.php]: reverted and reported: https://github.com/kalessil/phpinspectionsea/issues/1524

File: src/Roave/SecurityAdvisories/VersionConstraint.php (hunk 3 above)
Comment (orklah) [reply src/Roave/SecurityAdvisories/VersionConstraint.php]: fixed

**Pull request merged**